From b2e8a3e8eb2d14647c4ebf75a28f25e4ce4ac6cb Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Sat, 6 Jul 2019 07:54:44 +0000
Subject: [PATCH 136/187] HACK: arm64: dts: meson-gx: add ATF BL32 reserved
 memory region

Vendor firmware/uboot has an additional reserved region for BL32 trusted
firmware. If a board uses BL32 firmware, booting kernel without knowledge
of this region would cause an immediate kernel panic on SError Interrupt.

TODO: This should be enabled only for boards actually requiring it.

Signed-off-by: kszaq <kszaquitto@gmail.com>
---
 arch/arm64/boot/dts/amlogic/meson-gx.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
index 0ed78e10ce98..5a5cfaad3f5f 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -41,6 +41,12 @@
 			no-map;
 		};
 
+		/* 32 MiB reserved for ARM Trusted Firmware (BL32) */
+		secmon_reserved_bl32: secmon@5300000 {
+			reg = <0x0 0x05300000 0x0 0x2000000>;
+			no-map;
+		};
+
 		linux,cma {
 			compatible = "shared-dma-pool";
 			reusable;
-- 
2.17.1

